### PR TITLE
chore: update npm/vue to use latest test utils

### LINKED ISSUE #26575: In Cypress component testing, exposed values in a `script setup` component are not accessible on the component instance (AKA `wrapper.vm`)
### Current behavior

In Cypress component testing, exposed values in a `script setup` component are not accessible on the component instance (AKA `wrapper.vm`).

This is available when using `@vue/test-utils` directly in vitest.

image

### Desired behavior

Exposed values from a `script setup` component are accessible.

P.S. The type returned inside `Chainable` by `mount` is not easily accessible as it is defined inline, which makes it hard to use via an alias (`as`/`get`). It might be nice to turn it into an exported interface for ease of use?

### Test code to reproduce

https://github.com/segevfiner/cypress-test-utils-expose-issue

### Cypress Version

12.10.0

### Node version

v16.20.0

### Operating System

macOS 13.3.1

### Debug Logs
[cypress-debug.log](https://github.com/cypress-io/cypress/files/11313165/cypress-debug.log)

### Other

The same kind of test works in `vitest`.

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes https://github.com/cypress-io/cypress/issues/26575

### Additional details

Just updated the underlying `vue/test-utils` dependency to get some of the latest features, specifically, access variables declared and exposed via `defineExpose`.

I also fixed a bug where some of the Vue core code was incorrect bundled in the `cypress/vue` bundle. I fixed this by updating the `external` key in the rollup config. If you do `cd npm/vue && yarn build && wc -l dist/cypress-vue.esm-bundler.js` before and after, you'll see around 5k lines difference.

### Steps to test

You can run the new test if you want, but as long as CI is passing, everything is 💯 . This is just a dependency update.

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: npm/vue/cypress/component/setup/HelloWorld.cy.ts [hunk 1]
```diff
@@ -24,4 +24,15 @@ describe('HelloWorld', () => {
     .click()
     .should('contain', '1')
   })
+
+  it('exposes hello', () => {
```
Comment (mike-plummer) [inline npm/vue/cypress/component/setup/HelloWorld.cy.ts]: Should this be named more like "exposes `count` from `defineExpose`"?

File: cli/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,11 @@ _Released 05/09/2023 (PENDING)_
 
 - Fixed an issue in Electron where devtools gets out of sync with the DOM occasionally. Addresses [#15932](https://github.com/cypress-io/cypress/issues/15932).
 
+
+**Dependency Updates:**
+
+- Upgraded [`@vue/test-utils`](https://www.npmjs.com/package/vue) from `2.0.2` to `2.3.2`. Addresses [#26575](https://github.com/cypress-io/cypress/issues/26575).
```
Comment (mike-plummer) [inline cli/CHANGELOG.md]: Should this link point to https://www.npmjs.com/package/@vue/test-utils?

**Review by mike-plummer — COMMENTED** [review]
LGTM on the code side, just a couple minor tweaks needed

File: cli/CHANGELOG.md (hunk 2 above)
Comment (lmiller1990) [reply cli/CHANGELOG.md]: Yep, done

File: npm/vue/cypress/component/setup/HelloWorld.cy.ts (hunk 1 above)
Comment (lmiller1990) [reply npm/vue/cypress/component/setup/HelloWorld.cy.ts]: Fixed

**Review by mike-plummer — APPROVED** [review]

**Review by warrensplayer — APPROVED** [review]

**Review by astone123 — APPROVED** [review]

**Pull request merged**